### PR TITLE
Making configurable what method is used when adding k/v to spans

### PR DESCRIPTION
**Pull request by mapno**
#### PR Description 

Making configurable what method is used when adding k/v to spans. Options are `update`, `insert` and `upsert`

#### Which issue(s) this PR fixes 

Fixes https://github.com/grafana/agent/issues/795

#### PR Checklist

- [ ] CHANGELOG updated 
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: docs/configuration/tempo-config.md [hunk 1]
```diff
@@ -126,11 +126,17 @@ automatic_logging:
...
-# configs have their __address__ matched against the ip on incoming spans. If a
-# match is found then relabeling rules are applied.
-scrape_configs:
-  - [<scrape_config>]
+prom_sd:
+  # A list of prometheus scrape configs.  Targets discovered through these scrape
+  # configs have their __address__ matched against the ip on incoming spans. If a
+  # match is found then relabeling rules are applied.
+  scrape_configs:
+    - [<scrape_config>]
```
Comment (rfratto) [inline docs/configuration/tempo-config.md:134]: imo I'd prefer not nesting scrape_configs so it feels similar to the hierarchy used by logs / metrics. Can `sd_operation_type` be another top-level key maybe?

Comment (mapno) [reply docs/configuration/tempo-config.md:134]: Yeah, that's the other option I considered. I chose this approach for consistency with other processors' configs, but I'm not a huge fan of another breaking change. I'll change it 👍

Comment (mapno) [reply docs/configuration/tempo-config.md]: Changed in ea2ccc0b279d79aad4855954debe2c0d2c8d5821

File: pkg/tempo/config.go [hunk 2]
```diff
@@ -444,9 +445,14 @@ func (c *InstanceConfig) otelConfig() (*config.Config, error) {
 	processors := map[string]interface{}{}
 	processorNames := []string{}
 	if c.ScrapeConfigs != nil {
+		opType := promsdprocessor.OperationTypeUpsert
```
Comment (joe-elliott) [inline pkg/tempo/config.go:448]: is this validated anywhere?

Comment (mapno) [reply pkg/tempo/config.go:448]: Added in ba939c7f

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**